DynamoDbEncryption/runtimes/rust/examples/searchableencryption/beacon_styles_searchable_encryption.rs

```diff
@@ -0,0 +1,406 @@
+            ]),
+        ),
+        (
+            "veggies".to_string(),
+            AttributeValue::Ss(vec![
+                "beans".to_string(),
+                "carrots".to_string(),
+                "celery".to_string(),
+            ]),
+        ),
+        (
+            "work_type".to_string(),
+            AttributeValue::S("small".to_string()),
+        ),
+    ]);
+
+    // 9. Create item two, specifically with "dessert == fruit", and "fruit not in basket".
+    let item2 = HashMap::from([
+        ("work_id".to_string(), AttributeValue::S("2".to_string())),
+        (
+            "inspection_date".to_string(),
+            AttributeValue::S("2023-06-13".to_string()),
+        ),
+        (
+            "dessert".to_string(),
+            AttributeValue::S("orange".to_string()),
+        ),
+        ("fruit".to_string(), AttributeValue::S("orange".to_string())),
+        (
+            "basket".to_string(),
+            AttributeValue::Ss(vec![
+                "strawberry".to_string(),
+                "blueberry".to_string(),
+                "blackberry".to_string(),
+            ]),
+        ),
+        (
+            "veggies".to_string(),
+            AttributeValue::Ss(vec![
+                "beans".to_string(),
+                "carrots".to_string(),
+                "peas".to_string(),
+            ]),
+        ),
+        (
+            "work_type".to_string(),
+            AttributeValue::S("large".to_string()),
+        ),
+    ]);
+
+    // 10. Create a new AWS SDK DynamoDb client using the DynamoDb Config above
+    let dynamo_config = aws_sdk_dynamodb::config::Builder::from(&sdk_config)
+        .interceptor(DbEsdkInterceptor::new(encryption_config))
+        .build();
+    let ddb = aws_sdk_dynamodb::Client::from_conf(dynamo_config);
+
+    // 11. Add the two items
+    ddb.put_item()
+        .table_name(ddb_table_name)
+        .set_item(Some(item1.clone()))
+        .send()
+        .await?;
+
+    ddb.put_item()
+        .table_name(ddb_table_name)
+        .set_item(Some(item2.clone()))
+        .send()
+        .await?;
+
+    // 12. Test the first type of Set operation :
+    // Select records where the basket attribute holds a particular value
+    let expression_attribute_values = HashMap::from([(
+        ":value".to_string(),
+        AttributeValue::S("banana".to_string()),
+    )]);
+
+    let scan_response = ddb
+        .scan()
+        .table_name(ddb_table_name)
+        .filter_expression("contains(basket, :value)")
+        .set_expression_attribute_values(Some(expression_attribute_values.clone()))
+        .send()
+        .await?;
+
+    let attribute_values = scan_response.items.unwrap();
+    // Validate only 1 item was returned: item1
+    assert_eq!(attribute_values.len(), 1);
+    let returned_item = &attribute_values[0];
+    // Validate the item has the expected attributes
+    assert_eq!(returned_item["work_id"], item1["work_id"]);
+
+    // 14. Test the second type of Set operation :
```

Comment:
Nit:
```suggestion
    // 13. Test the second type of Set operation :
```
Ignoring line number comments in all files from here